### DISCUSSION TITLE
hrp convert .har转换用例pytest报错

### DISCUSSION THREAD

**Discussion by hans-jp** [httprunner/httprunner#1672] [2 - 答疑（Q&A）]
执行转换命令hrp convert Untitled.har --to-pytest --from-har
报错：
11:56AM INF exec command cmd="C:\\windows\\system32\\cmd.exe"
Fatal Python error: Py_Initialize: unable to load the file system codec
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00041150 (most recent call first):
11:56AM ERR exec command failed error="exit status 0xc0000409"
11:56AM WRN pip is not available
11:56AM ERR python3 venv is not ready error="pip install httprunner==v4.3.5 failed: pip is not available: exit status 0xc0000409: prepare python3 venv failed"
Error: pip install httprunner==v4.3.5 failed: pip is not available: exit status 0xc0000409: prepare python3 venv failed
11:56AM WRN hrp exit code=9

**Comment by hans-jp** [comment]
原因：hrp版本为4.3.5，python初始环境为python3.6，后替换为python3.10，由于执行hrp convert Untitled.har --to-pytest --from-har的命令是在创建demo项目的环境下执行的，所以当时的环境依旧是python3.6。

解决：手动删除项目后，重新创建项目，在项目下重新执行hrp convert Untitled.har --to-pytest --from-har命令，发现报错1```
missing dependency tool: black
install black manually and try again:
$ pip install black
```
继续解决black依赖问题，执行pip install black命令，又报错2：
`Fatal error in launcher: Unable to create process using '"C:\Users\***\AppData\Local\Programs\Python\Python310\python.exe"  "D:\***\Python\Python310\Scripts\pip.exe" install black': ???????????`
接着解决pip问题，执行python -m pip install --upgrade pip，还报错3：There was an error checking the latest version of pip.
无奈手动下载pip包，执行python setup.py install重新安装pip，检查安装成功后，
重试解决black依赖问题，执行python -m pip install black命令，执行成功；
最后进入新项目下执行hrp convert Untitled.har --to-pytest --from-har命令，执行成功，.har成功转为.py(此时会生成3个文件，依次为Untitled_test.json、_init_.py、Untitled_test_test.py)

**Reply by YaShuaiLI** [reply]
我今天也遇到了这个问题，请问你解决了吗

**Reply by hans-jp** [reply]
已经解决了，你可以参考一下我的第一个回答

**Reply by YaShuaiLI** [reply]
我按照你的步骤来操作一下  最近没看这个框架
